extras/spiffs/component.mk

```diff
@@ -1,7 +1,30 @@
 # Component makefile for extras/spiffs
 
+# If spiffs is configured as SINGLETON it must be configured in compile time.
+SPIFFS_SINGLETON ?= 1
+
 SPIFFS_BASE_ADDR ?= 0x300000
 SPIFFS_SIZE ?= 0x100000
+SPIFFS_LOG_PAGE_SIZE ?= 256
+SPIFFS_LOG_BLOCK_SIZE ?= 8192
+SPIFFS_ERASE_BLOCK_SIZE ?= 4096
```

Comment:
I would actually prefer not having ERASE_BLOCK_SIZE configurable.  As far as i know the esp hal only supports 4k erases.